src/components/Cluster/ClusterDetail/AddNodePool/AddNodePoolAZSelection.tsx

```diff
@@ -135,70 +135,74 @@ const AddNodePoolAZSelection: React.FC<IAddNodePoolAZSelectionProps> = ({
   return (
     <div {...rest}>
       <AZSelectionLabel>Availability Zones selection</AZSelectionLabel>
-      <StyledPanel
-        expanded={value === AvailabilityZoneSelection.Automatic}
-        onToggle={onToggleFakeCallback}
-      >
-        <AddNodePoolAZSelectionCheckbox
-          onChange={onChange}
-          value={value}
-          npID={npID}
-          label='Automatic'
-          type={AvailabilityZoneSelection.Automatic}
-        />
-        <StyledPanelCollapse>
-          <AZSelectorWrapper>
-            <p>Number of availability zones to use:</p>
-            <AvailabilityZonesParser
-              min={minNumOfZones}
-              max={maxNumOfZones}
-              defaultValue={defaultNumOfZones}
-              zones={allZones}
-              updateAZValuesInParent={onUpdateZones(
-                AvailabilityZoneSelection.Automatic
-              )}
-              isLabels={false}
-            />
-          </AZSelectorWrapper>
-          <p>{automaticAZSelectionMessage}</p>
-        </StyledPanelCollapse>
-      </StyledPanel>
-      <StyledPanel
-        expanded={value === AvailabilityZoneSelection.Manual}
-        onToggle={onToggleFakeCallback}
-      >
-        <AddNodePoolAZSelectionCheckbox
-          onChange={onChange}
-          value={value}
-          npID={npID}
-          label='Manual'
-          type={AvailabilityZoneSelection.Manual}
-        />
-        <StyledPanelCollapse>
-          <p>
-            You can select up to {maxNumOfZones} availability zones to make use
-            of.
-          </p>
-          <AZSelectorWrapper>
-            <ManualAZSelector>
+      {maxNumOfZones !== undefined && maxNumOfZones > 0 && (
```

Comment:
We have a default value set up for this
https://github.com/giantswarm/happa/blob/53c22fd108c74cf2f1aa3fef35821c6f03908bc2/src/components/Cluster/ClusterDetail/AddNodePool/AddNodePoolAZSelection.tsx#L241

So when the `maxNumOfZones` prop is not assigned any value, it takes the default value declared above. This means that the value is never `undefined`. However, we have to define it as optional in the props type definition in order to not have to specify it when using the component.

Long story short, we don't have to check for `undefined` here, and it's safe to use type casting.

```suggestion
      {(maxNumOfZones as number) > 0 && (
```